Source/JavaScriptCore/runtime/PropertyTable.h

```diff
@@ -532,13 +535,24 @@ inline void PropertyTable::reinsert(Index* indexVector, Entry* table, const Valu
     // Used to insert a value known not to be in the table, and where
     // we know capacity to be available.
     ASSERT(canInsert(entry));
-    FindResult result = findImpl(indexVector, table, entry.key());
-    ASSERT(result.offset == invalidOffset);
-    ASSERT(result.entryIndex == EmptyEntryIndex);
+
+    unsigned hash = IdentifierRepHash::hash(entry.key());
+    unsigned indexMask = m_indexMask;
+    unsigned probeCount = 0;
+    unsigned index = hash & indexMask;
+
+    // Reinsert must not conflict with the keys since all entries are existing ones.
+    // Plus, there is no deleted entries too. We should just check emptyness, that's it.
+    while (true) {
+        if (indexVector[index] == EmptyEntryIndex)
```

Comment:
Can we add an assert that none of the keys are going to be deleted? IIUC, reinsert is only called when growing the property table.

Reply:
Done